/** (C) Copyright 2013, Applied Physical Sciences Corp., A General Dynamics Company
 **
 ** Gravity is free software; you can redistribute it and/or modify
 ** it under the terms of the GNU Lesser General Public License as published by
 ** the Free Software Foundation; either version 3 of the License, or
 ** (at your option) any later version.
 **
 ** This program is distributed in the hope that it will be useful,
 ** but WITHOUT ANY WARRANTY; without even the implied warranty of
 ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 ** GNU Lesser General Public License for more details.
 **
 ** You should have received a copy of the GNU Lesser General Public
 ** License along with this program;
 ** If not, see <http://www.gnu.org/licenses/>.
 **
 */

syntax = "proto2";
option optimize_for = SPEED;
option java_outer_classname = "GravityMetricsDataContainer";
option java_package = "com.aphysci.gravity.protobuf";

package gravity;

message GravityMetricsPB
{
	optional string dataProductID = 1;
	enum MessageType
	{
		PUBLICATION = 0;
		SUBSCRIPTION = 1;
		REQUEST = 2;
		RESPONSE = 3;
	}
	optional MessageType messageType = 2;
	repeated uint64 startTime = 3 [packed=true];
	repeated uint64 endTime = 4 [packed=true];
	repeated uint32 numBytes = 5 [packed=true];
	repeated uint32 numMessages = 6 [packed=true];
}

message GravityMetricsDataPB
{
	optional string componentID = 1;
	optional string ipAddress = 2;
	repeated GravityMetricsPB metrics = 3;
}
